app/lib/katello/resources/registry.rb

```diff
@@ -12,22 +12,38 @@ def self.logger
 
         def self.get(path, headers = {:accept => :json})
           logger.debug "Sending GET request to Registry: #{path}"
-          client = RegistryResource.rest_client(Net::HTTP::Get, :get, path)
+          client = RegistryResource.load_class.rest_client(Net::HTTP::Get, :get, path)
           client.get(headers)
         end
       end
 
       class RegistryResource < HttpResource
-        if SETTINGS[:katello][:container_image_registry]
-          cfg = SETTINGS[:katello][:container_image_registry]
-          url = cfg[:crane_url]
-          uri = URI.parse(url)
-          self.prefix = uri.path
-          self.site = "#{uri.scheme}://#{uri.host}:#{uri.port}"
-          self.ca_cert_file = cfg[:crane_ca_cert_file]
-        end
+        extend Concerns::SmartProxyExtensions
 
         class << self
+          def load_class
+            container_config = SETTINGS.dig(:katello, :container_image_registry)
+            registry_url = nil
+
+            if container_config
+              crane_url = container_config[:crane_url]
+              pulp_registry_url = container_config[:pulp_registry_url]
+              registry_url = crane_url if crane_url
+              # pulp 3 acts as its own registry
+              if pulp_registry_url && content_pulp3_support?(::Katello::DockerBlob::CONTENT_TYPE)
+                registry_url = pulp_registry_url
+              end
```

Comment:
Would this be clearer? 
`registry_url = content_pulp3_support?(::Katello::DockerBlob::CONTENT_TYPE) ?  pulp_registry_url: crane_url`